### PR TITLE
Update MCPrio/Prio order in Haskell, prove bindNotification_invs'

### PR DESCRIPTION
**Pull request by pierzchalski**

### CODE REVIEW

**Comment by pierzchalski** [comment]
[Testboard](https://bamboo.ts.data61.csiro.au/browse/L4V-TESTBOARD-648)

File: proof/refine/ARM/Tcb_R.thy [hunk 1]
```diff
@@ -2414,35 +2414,17 @@ lemma bindNotification_invs':
...
-  done *)
+  apply (wpsimp wp: set_ntfn_valid_pspace' sbn_sch_act' sbn_valid_queues valid_irq_node_lift
+                    setBoundNotification_ct_not_inQ valid_bound_ntfn_lift
+                    untyped_ranges_zero_lift irqs_masked_lift
+              simp: cteCaps_of_def)
+  apply (frule(1) ntfn_ko_at_valid_objs_valid_ntfn'[OF _ valid_pspace_valid_objs'])
+  apply (clarsimp simp: obj_at'_def pred_tcb_at'_def valid_ntfn'_def projectKOs o_def
+                        global'_no_ex_cap
+                 split: ntfn.splits)
+  done
```
Comment (corlewis) [inline proof/refine/ARM/Tcb_R.thy:2427]: Nice job improving this proof!

**Review by corlewis — APPROVED** [review]
Just confirming, am I correct in saying that the MCPrio/Prio change does not affect any proofs, but makes it a bit easier to think about since now the parameters are in a consistent order?

**Comment by pierzchalski** [comment]
> 
> 
> Just confirming, am I correct in saying that the MCPrio/Prio change does not affect any proofs, but makes it a bit easier to think about since now the parameters are in a consistent order?

Yep: because of the way `tcb_inv_wf'` is defined, changing the invocation arg order makes `tcb_inv_wf'` correct.

**Review by mktnk3 — APPROVED** [review]
looks good. I guess we can merge this when rebasing is done.

**Comment by pierzchalski** [comment]
[post-rebase testboard](https://bamboo.ts.data61.csiro.au/browse/L4V-TESTBOARD-655)

**Pull request merged**